### ISSUE TITLE
Quoted operators not supported in [ ]

### ISSUE THREAD

**Issue by SpecLad** [koalaman/shellcheck#492]
```
   1  if [ abc '<' def ]; then :; fi
      ^––SC1009 The mentioned parser error was in this if expression.
         ^––SC1073 Couldn't parse this test expression.
               ^––SC1072 Expected test to end here. Fix any mentioned problems and try again.
```
`\<` works, as does `test abc '<' def`.

**Comment by koalaman** [comment]
Fixed in ffb9578. It's not a proper shell token parsing, but it fixes these basic cases.

**Comment by pikrzysztof** [comment]
It fails for "!"

Please see the screenshot

![screenshot from 2018-07-10 18-28-08](https://user-images.githubusercontent.com/3964215/42526905-1a4aee06-846f-11e8-917e-15a287dadca8.png)
